packages/apps/plugins/plugin-deck/src/components/DeckLayout.tsx

```diff
@@ -295,69 +294,73 @@ export const DeckLayout = ({
   const { t } = useTranslation(DECK_PLUGIN);
   const { graph } = useGraph();
 
-  // TODO(wittjosiah): Finding nodes in the graph should probably not be done at the top-level of layout.
-  //   This likely is causing the whole layout to re-render more than necessary.
-  const activeParts: ActiveParts = isActiveParts(location.active)
-    ? Object.keys(location.active).length < 1
-      ? { sidebar: NAV_ID }
-      : location.active
-    : { sidebar: NAV_ID, main: [location.active].filter(Boolean) as string[] };
-  const sidebarSlug = firstSidebarId(activeParts);
-  const sidebarNode = useNode(graph, sidebarSlug);
-  const sidebarAvailable = sidebarSlug === NAV_ID || !!sidebarNode;
-  const fullScreenSlug = firstFullscreenId(activeParts);
+  const sidebarNodeId = useMemo(() => firstIdInPart(layoutParts, 'sidebar'), [layoutParts]);
+  const sidebarAvailable = useMemo(() => sidebarNodeId === NAV_ID, [sidebarNodeId]);
```

Comment:
fwiw my understanding is that memoizing things like this which take a negligible amount of time to compute doesn't actually save any time because the memoization needs to do an equality check at minimum as well

Reply:
I was seeing an infinite re-render issue until I memoised. Didn't narrow in on the loop precisely, but we're going to be slicing this up soon?